bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/PatientSearchParameters.java

```diff
@@ -55,7 +56,7 @@ public PatientSearchParameters(RequestContext context) {
         }
         this.setAddressFieldValue(context.getParameter("addressFieldValue"));
         Map parameterMap = context.getRequest().getParameterMap();
-        this.setAddressSearchResultFields((String[]) parameterMap.get("addressSearchResultsConfig"));
+        this.setAddressSearchResultFields(!Arrays.toString((Object[]) parameterMap.get("addressSearchResultsConfig")).equals("[{}]") ? (String[]) parameterMap.get("addressSearchResultsConfig") : null);
```

Comment:
Would be better to extract this to a private method which indicates what is happening. E.g: `getAddressConfigIfExists()`

Reply:
done